### PR TITLE
Change the way to get the price without discounts

### LINKED ISSUE #16163: If a discount from one unit is created, discounts on subsequent quantities will always include a discount of the first one unit
**Describe the bug**

**To Reproduce**
1. create multiple specific prices rules (percentage) 
2. one of them should be from a quantity equal to one product 
3. Product settings -> Display discounted price must be enabled 
4. As a result, we will get an incorrect calculation in the table (see the screenshot)

**Screenshots**

**Additional information**
PrestaShop version: 1.7.*
PHP version: 7.1

**fixed ticket**
#15100

### PR DESCRIPTION
**Pull request by levyn**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | If a discount from one unit is created, discounts on subsequent quantities will always include a discount of the first one unit. To avoid this, it is necessary to use the cost in calculations without taking into account previous discounts.
| Type?         | bug fix 
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  Fixes #16163
| How to test?  | 1. create multiple specific prices rules (percentage)<br>2. one of them should be from a quantity equal to one product<br>3. Product settings -> Display discounted price must be enabled<br>As a result, we will get an incorrect calculation in the table - http://joxi.ru/KAxplPgHZE6NYA

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15100)

### CODE REVIEW

**Comment by marionf** [comment]
Hello @levyn 

We have already issues on this subject:
https://github.com/PrestaShop/PrestaShop/issues/15835
https://github.com/PrestaShop/PrestaShop/issues/15866
Please, take a look on them and tell us if your PR fix one of them

Otherwise, could you please create an issue and describe the steps to reproduce the issue ?
We need to know the price tax excl of the product, the VAT amount, the percentage of each specific price created and if you display prices with or without taxes in front-office

Thank you !

**Comment by levyn** [comment]
Hi @marionf 
No, this correction only concerns the current problem. Other issues  don't even intersect here.
But, I am proposing corrections for two other issues in this PR - [#16138](https://github.com/PrestaShop/PrestaShop/pull/16138)

**Comment by marionf** [comment]
ok @levyn 

So, could you please create an issue to describe the steps to reproduce the problem and link it to this PR in "fixed ticket" section ?

**Review by mBangiev — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -514,7 +514,7 @@ protected function assignPriceAndTax()
             }
         }
 
-        $product_price = $this->product->getPrice(Product::$_taxCalculationMethod == PS_TAX_INC, $id_product_attribute);
+        $product_price = $this->product->getPrice(Product::$_taxCalculationMethod == PS_TAX_INC, $id_product_attribute, 6, null, false, false);
```
Comment (Progi1984) [inline controllers/front/ProductController.php:517]: Why 6 ? There is no constant for this ?

**Review by Progi1984 — COMMENTED** [review]
Feedback

File: controllers/front/ProductController.php (hunk 1 above)
Comment (levyn) [reply controllers/front/ProductController.php:517]: No, there isn't.
The original method contain this value without constant, as you can see at the picture.
getPrice

Comment (Progi1984) [reply controllers/front/ProductController.php:517]: OK. Thanks

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
Hello @levyn  The PR correctly fix #16163 as you can see : 

**Before the PR :** 
capture d'écran_18

**After the PR :** 
capture d'écran_19

Thanks @levyn

**Comment by PierreRambaud** [comment]
Thanks @levyn

**Pull request merged**